pkg/testgridanalysis/testgridconversion/to_raw_data_test.go

```diff
@@ -0,0 +1,318 @@
+		assert.NotNil(t, result.TestResults[s], "Expected non nil test result for %s", s)
+	}
+
+}
+
+func TestRegexExplicit(t *testing.T) {
+
+	// explicit test cases and verifications
+	testName1 := "\"Installing \"Red Hat Integration - 3scale\" operator in test-ieesa.Installing \"Red Hat Integration - 3scale\" operator in test-ieesa Installs Red Hat Integration - 3scale operator in test-ieesa and creates 3scale Backend Schema operand instance\""
+	testName1Clean := "\"Installing \"Red Hat Integration - 3scale\" operator in test-random.Installing \"Red Hat Integration - 3scale\" operator in test-random Installs Red Hat Integration - 3scale operator in test-random and creates 3scale Backend Schema operand instance\""
+
+	testName2 := "\"Installing \"Red Hat Integration - 3scale\" operator in test-jopkv.Installing \"Red Hat Integration - 3scale\" operator in test-jopkv \"after all\" hook for \"Installs Red Hat Integration - 3scale operator in test-jopkv and creates 3scale Backend Schema operand i (...)\""
+	testName2Clean := "\"Installing \"Red Hat Integration - 3scale\" operator in test-random.Installing \"Red Hat Integration - 3scale\" operator in test-random \"after all\" hook for \"Installs Red Hat Integration - 3scale operator in test-random and creates 3scale Backend Schema operand i (...)\""
+
+	testName1 = cleanTestName(testName1)
+	assert.Equal(t, testName1, testName1Clean, "TestName1Clean did not match, %s", testName1)
+
+	testName2 = cleanTestName(testName2)
+	assert.Equal(t, testName2, testName2Clean, "TestName2Clean did not match, %s", testName2)
+}
+
+func TestRegexGeneric(t *testing.T) {
+
+	// if we change the input data we might have to reconsider the hardcoded fixCount validation
+	// for the example data we are working with when there is a random entry to be replaced
+	// it shows up in 3 locations, this is verifying all 3 are getting updated
+	expectedRandomReplacements := 3
+
+	// get our list of tests
+	testStrings := buildTestNameSamples()
+
+	// get the count for the names we expect will get changes
+	validateCount := len(testStrings)
+
+	// add in the ones we don't expect to get changed after we have the count of original ones
+	// intentional non matching replace with no trailing a-z after test-
+	nonMatchString := "\"Installing \"Red Hat Integration - 3scale\" operator in test-.Installing \"Red Hat Integration - 3scale\" operator in test- Installs Red Hat Integration - 3scale operator in test- and creates 3scale Backend Schema operand instance\""
+	testStrings = append(testStrings, nonMatchString)
+
+	// will not trigger the first match for the starts with
+	skipString := "\"Doesn'tStartWith Installing \"Red Hat Integration - 3scale\" operator in test-.Installing \"Red Hat Integration - 3scale\" operator in test- Installs Red Hat Integration - 3scale operator in test- and creates 3scale Backend Schema operand instance\""
+	testStrings = append(testStrings, skipString)
+
+	fixedNames := 0
+
+	for _, s := range testStrings {
```

Comment:
There's a very common sub-test pattern used in go here that's probably worth playing with this PR. It would replace the testStrings below (can probably cut down on those to just a handful), as well as these additional testStrings you're adding. You define a temporary struct that represents a test case, including a name, the inputs, and the expected output. Then you define an array of those, iterate them, and use go's sub-test functionality to turn each into a separate sub-test output. Here's a good example: https://github.com/openshift/origin/blob/master/pkg/synthetictests/duplicated_events_test.go#L13. This blog explains a little too: https://medium.com/@sebdah/introduction-to-go-sub-tests-5fca2b9f6983

Reply:
This will make it a little easier to see inputs and expected output when looking over what's tested.

Reply:
Got it, thanks for the pointer.  Updated the tests